### PR TITLE
Add workflow to remind PR reviewers to also review the testing instructions.

### PR DESCRIPTION
**Pull request by alopezari**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a new GitHub Actions workflow that triggers every time a GitHub user is added as a reviewer to a PR. When this happens, a new automatically generated comment is added to the PR tagging the requested reviewers and suggesting them to review also the testing instructions, providing the [guide for writing high-quality testing instructions](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions) as reference.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

#### New comment is added for the first time in a PR where any reviewer is requested
1. Fork the WooCommerce repository.
2. Make sure you added more users to the repository so that you could tag them as reviewers in a PR.
3. Add this new workflow to the `.github/workflows/` path in the trunk branch of your forked WooCommerce repository. For example, you could create a new file at `.github/workflows/review-testing-instructions.yml` containing the changes of this PR.
4. Again in the forked repository, create a new branch to add a dummy change (such as a small change in a README.md file, for example).
5. Push the branch to remote and create a PR.
6. Add someone as the PR reviewer.
7. Check that the new workflow named `Remind reviewers to also review the testing instructions` starts running in the PR after a new reviewer was requested.
8. Once the new workflow has finished running, check that a new comment is added to the PR, with the following text:
```
Hi {REVIEWERS},

Apart from reviewing the code changes, please make sure to review the testing instructions as well.

You can follow this guide to find out what good testing instructions should look like:
https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions
```

Where {REVIEWERS} is a comma-separated list of the reviewers' usernames. For example:

Screenshot 2023-02-22 at 14 39 03

You can see a live example [here](https://github.com/alopezari/woocommerce/pull/3#issuecomment-1439993565).

#### Previous comment asking for reviewing testing instructions is updated in a PR where a new reviewer is requested in addition to the previous ones
1. Follow the testing instructions described above to get a new PR where the workflow added this comment after requesting a reviewer:
```
Hi {REVIEWERS},

Apart from reviewing the code changes, please make sure to review the testing instructions as well.

You can follow this guide to find out what good testing instructions should look like:
https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions
```
2. Request a new reviewer or remove an existing reviewer and request their review again.
3. Check that the new workflow named `Remind reviewers to also review the testing instructions` starts running in the PR after a new reviewer was requested.
4. Once the new workflow has finished running, check that the comment described in the first step is updated to include the new reviewers.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by alopezari** [comment]
I've seen the new workflow was executed 3 times. This isn't expected, could it be related to how the @woocommerce/atlas alias resolves to one person or another?

I'll investigate this. Will set the PR as `draft` in the meanwhile.

**Comment by jonathansadowski** [comment]
> I've seen the new workflow was executed 3 times

May I suggest an approach that updates the comment if it exists, otherwise creates a new one?  I *believe* that if you insert an html comment into the comment, it'll survive, but not be visible, which should give you something unique to search for an existing comment.

**Comment by alopezari** [comment]
> > I've seen the new workflow was executed 3 times
> 
> May I suggest an approach that updates the comment if it exists, otherwise creates a new one? I *believe* that if you insert an html comment into the comment, it'll survive, but not be visible, which should give you something unique to search for an existing comment.

Thanks for the idea @jonathansadowski! It makes perfect sense.

The main reason why I tagged reviewers in the comment is that I want to make sure that they saw the message, so by mentioning them in the comment, they would receive a notification.

I've tested in a separate PR if you can receive a notification if you're mentioned in a comment that was edited, but seems it doesn't work 🤔 

As I see it now, we could:
- Edit the comment if it was already posted -> new reviewers won't receive any notification.
- Remove the previous comment and create a new one -> all reviewers will receive a new notification.

In both cases, this message will be displayed only once.

What approach do you think would be better @jonathansadowski @psealock?

Thank you!

**Comment by alopezari** [comment]
I updated the workflow to check if there is a previous version of the comment before commenting and, in that case, updating it instead of creating a new one, as suggested [here](https://github.com/woocommerce/woocommerce/pull/36918#issuecomment-1440379190).

It updated successfully the [first comment](https://github.com/woocommerce/woocommerce/pull/36918#issuecomment-1440055183) created by the new workflow in this PR:

image

**Comment by alopezari** [comment]
Updated the testing instructions to cover also the new scenario and set the PR ready for review.

File: .github/workflows/review-testing-instructions.yml
Comment (tammullen) [inline .github/workflows/review-testing-instructions.yml]: We should pin to a SHA of this third party action 
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

File: .github/workflows/review-testing-instructions.yml
Comment (tammullen) [inline .github/workflows/review-testing-instructions.yml]: Same for this one to pin to SHA
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

**Review by tammullen — CHANGES_REQUESTED** [review]

File: .github/workflows/review-testing-instructions.yml
Comment (alopezari) [reply .github/workflows/review-testing-instructions.yml]: Thanks, @tammullen! I didn't know about this practice.

Updated it in a [new commit](https://github.com/woocommerce/woocommerce/pull/36918/commits/5052159c17f675f42d3d664fa319795ce198c9cf) and also applied the same change to other workflows where we used the same third-party actions.

I referenced both actions to their respective commit SHA of the v2 tag:
- [peter-evans/find-comment](https://github.com/peter-evans/find-comment/commit/034abe94d3191f9c89d870519735beae326f2bdb)
- [peter-evans/create-or-update-comment](https://github.com/peter-evans/create-or-update-comment/commit/67dcc547d311b736a8e6c5c236542148a47adc3d)

We can see the workflow keeps working as expected after this change:
image

**Comment by psealock** [comment]
@alopezari and @tammullen I've invited you to my fork as members so you can be pinged as reviewers. I think you'll need to accept because otherwise I can't add you. Cheers

image

**Comment by alopezari** [comment]
Hi @tammullen, @psealock!

Any news on this?

**Comment by tammullen** [comment]
I noticed on @psealock 's PR in the forked branch [it didn't run when I self-requested as reviewer.](https://github.com/psealock/woocommerce/pull/26)

@nigeljamesstevenson can you add yourself as a reviewer to this PR to see if the comment updates https://github.com/woocommerce/woocommerce/pull/36918#issuecomment-1440055183

**Comment by tammullen** [comment]
Thanks!
Self-requesting worked fine here so might just be forked branches it doesn't
Screenshot 2023-03-21 at 16 30 29

**Comment by alopezari** [comment]
> I noticed on @psealock 's PR in the forked branch [it didn't run when I self-requested as reviewer.](https://github.com/psealock/woocommerce/pull/26)

Yep, seems the [workflow is missing](https://github.com/psealock/woocommerce/tree/trunk/.github/workflows) in the trunk branch of that fork, I don't know if it could be related. It's in the [commit](https://github.com/psealock/woocommerce/blob/4d0a785d21fdf86af0d8bf8425b411732cb5ee6a/.github/workflows/review-testing-instructions.yml) though.

I can't see if the action was run there.

**Comment by psealock** [comment]
>I can't see if the action was run there.

I keep Actions disabled normally. Once I enabled them and added @alopezari the action ran and the comment was generated.

image

**Review by psealock — APPROVED** [review]
This is testing well and code looks good. Thanks @alopezari.

I'm curious what the reception will be and am looking forward to seeing if this improves code reviews.

**Comment by alopezari** [comment]
Thanks, @psealock! 

We've already seen a noticeable improvement in testing instructions recently thanks to other initiatives. Hope this one keeps moving us in that direction 🤞 It would be amazing if it also serves to detect other potential gaps in the changes submitted in PRs.

Hi @tammullen! I think we need your approval again after the latest changes you required:
image

Once this is addressed, we will be able to merge this PR. Please let me know if there is anything else we could do to improve it!

**Review by tammullen — APPROVED** [review]

**Pull request merged**
